builtin/logical/mysql/secret_creds.go

```diff
@@ -59,27 +62,51 @@ func (b *backend) secretCredsRevoke(
 		return nil, err
 	}
 
-	// Start a transaction
-	tx, err := db.Begin()
+	// Get the role
+	pathParts := strings.Split(req.Path, "/")
```

Comment:
You should put the role name into the secret internal data at creation time and read it out here rather than doing path munging.